admin_site/system/views.py

```diff
@@ -1874,3 +1895,50 @@ def get_context_data(self, **kwargs):
                 pc[pn] = pv
             pcs.append(pc)
         return pcs
+
+
+class ImageVersionsView(SiteMixin, SuperAdminOrThisSiteMixin, ListView):
+    """Image Versions are are sorted by major versions in the view
+    """
+
+    template_name = 'system/site_image_versions.html'
+    model = ImageVersion
+    context_object_name = 'image_versions'
+    selection_class = ImageVersion
+    class_display_name = 'image_version'
+
+    def get_context_data(self, **kwargs):
+        context = super(ImageVersionsView, self).get_context_data(**kwargs)
+
+        site_uid = self.kwargs.get('site_uid')
+        site_obj = Site.objects.get(uid=site_uid)
+        site_allowed = False
+        if site_obj.last_version:
+            site_allowed = True
+
+        context["site_allowed"] = site_allowed
+
+        versions = ImageVersion.objects.all()
+
+        major_versions_set = set()
+        for minor_version in versions:
+            major_versions_set.add(minor_version.img_vers[:1])
+        major_versions_list = list(major_versions_set)
+        major_versions_list.sort(reverse=True)
+
+        context["major_versions"] = major_versions_list
+
+        url_ref_vers = self.kwargs.get(
+            'major_version',
+            major_versions_list[0]
```

Comment:
I remark that this line makes the administration system crash if no versions are configured:

![Screenshot_2020-02-14 IndexError at site magenta image-versions ](https://user-images.githubusercontent.com/45562185/74542960-dcdc0a80-4f44-11ea-8dd6-edea845d7208.png)

Reply:
Check